code/game/machinery/jukebox.dm

```diff
@@ -25,31 +19,125 @@
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 10
 	active_power_usage = 100
+	circuit = /obj/item/weapon/electronics/circuitboard/jukebox
 
-	var/playing = 0
-	var/volume = 20 // TODO: Adjustable volume
-	var/range = 7
-
-	var/sound_id
-	var/datum/sound_token/sound_token
+	// Vars for hacking
+	var/datum/wires/jukebox/wires = null
+	var/hacked = FALSE // Whether to show the hidden songs or not
+	var/freq = 0 // Currently no effect, will return in phase II of mediamanager.
 
-	var/datum/track/current_track
-	var/list/datum/track/tracks
+	var/loop_mode = JUKEMODE_PLAY_ONCE			// Behavior when finished playing a song
+	var/max_queue_len = 3						// How many songs are we allowed to queue up?
+	var/datum/track/current_track				// Currently playing song
+	var/list/datum/track/queue = list()			// Queued songs
+	var/list/datum/track/tracks = list()		// Available tracks
+	var/list/datum/track/secret_tracks = list() // Only visible if hacked
 
-/obj/machinery/media/jukebox/Initialize()
+/obj/machinery/media/jukebox/New()
```

Comment:
Why Initialize to new?

Reply:
Weird fuckery with wires and areas. I think it's something about our initialize handling, same reason quite a bit of things are fucky when lateloaded (like pipes).